talentprotocol/web-dapp
# feat: moderator role

## Summary

This PR;
- adds a moderator role to users
- adds a rake task to set a user as a moderator
- allows moderators to approve/reject talents
- shows who approved the talent to admins

![Screenshot 2022-10-29 at 12 39 31](https://user-images.githubusercontent.com/33064059/198829406-cb361717-e502-476d-9599-f6863929b527.png)

## Notion link

https://talentprotocol.notion.site/Develop-Moderator-Role-2f254e51076c4910b2f086319a4cd161

## How to test?

- Run rake task to set a `basic` user to be moderator with `rails "users:set_user_as_moderator[user_id]"`
- Sign in as user
- Go to a talent profile that has a `waiting_for_approval` profile type
- You should be able to see the "Approve" or "Reject" buttons and be able to click them
- Sign out
- Sign in as an admin
- Go back to the same profile and you should see who approved the talent
